sensing/src/main/java/com/google/android/sensing/capture/CaptureUtil.kt

```diff
@@ -16,43 +16,25 @@
 
 package com.google.android.sensing.capture
 
-import android.content.Context
-import android.content.res.ColorStateList
-import androidx.camera.core.TorchState
-import androidx.core.content.ContextCompat
-import com.google.android.fitbit.research.sensing.common.libraries.camera.Camera2InteropSensor
-import com.google.android.material.floatingactionbutton.FloatingActionButton
-import com.google.android.sensing.R
-import com.google.android.sensing.model.CaptureType
-import com.google.android.sensing.model.InternalSensorType
+import java.io.BufferedOutputStream
+import java.io.FileOutputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
 
 class CaptureUtil {
   companion object {
-    fun toggleFlashWithView(
-      camera: Camera2InteropSensor,
-      toggleFlashFab: FloatingActionButton,
-      context: Context
-    ) {
-      val c = camera.cameraXSensor.camera
-      if (c!!.cameraInfo.torchState.value == TorchState.ON) {
-        // Turn off flash
-        c.cameraControl.enableTorch(false)
-        toggleFlashFab.setImageResource(R.drawable.flashlight_off)
-        toggleFlashFab.backgroundTintList =
-          ColorStateList.valueOf(ContextCompat.getColor(context, android.R.color.black))
-      } else {
-        // Turn on flash
-        c.cameraControl.enableTorch(true)
-        toggleFlashFab.setImageResource(R.drawable.flashlight_on)
-        toggleFlashFab.backgroundTintList =
-          ColorStateList.valueOf(ContextCompat.getColor(context, R.color.colorPrimary))
-      }
-    }
-
-    fun sensorsInvolved(captureType: CaptureType): List<InternalSensorType> {
-      return when (captureType) {
-        CaptureType.VIDEO_PPG -> listOf(InternalSensorType.CAMERA)
-        CaptureType.IMAGE -> listOf(InternalSensorType.CAMERA)
+    fun zipDirectory(sourceDirPath: Path, zipPath: Path) {
+      ZipOutputStream(BufferedOutputStream(FileOutputStream(zipPath.toFile()))).use { zipOut ->
+        Files.walk(sourceDirPath)
+          .filter { path -> !Files.isDirectory(path) } // Exclude directories from the initial walk
+          .forEach { path ->
+            val zipEntry = ZipEntry(sourceDirPath.relativize(path).toString())
```

Comment:
i kinda like the idea of inlining this variable. but this is optional - try it and see if you think it's more succint. up to you!